src/System.Configuration.ConfigurationManager/tests/System/Configuration/UrlPathTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Configuration;
+using System.IO;
+using Xunit;
+
+namespace System.ConfigurationTests
+{
+    public class UrlPathTests
+    {
+        private const string TEST_FILE_NAME = "TestFileForUrlPathTests.txt";
+        private const string TEST_DIRECTORY_PATH = "C:\\Directory";
+        private const string TEST_SUBDIRECTORY_PATH = TEST_DIRECTORY_PATH + "\\SubDirectory";
+
+        public const string TEST_DIRECTORY_PATH_WITH_BACKSLASH = TEST_DIRECTORY_PATH + "\\";
+        public const string TEST_SUBDIRECTORY_PATH_WITH_BACKSLASH = TEST_SUBDIRECTORY_PATH + "\\";
+
+        [Fact]
+        public void GetDirectoOrRootName_Null()
+        {
+            string test = UrlPath.GetDirectoryOrRootName(null);
+            Assert.Equal(null, null);
+        }
+
+        [Fact]
+        public void GetDirectoryOrRootName_NotDIrectoryOrRoot()
+        {
+            string test = UrlPath.GetDirectoryOrRootName("Hello");
+            Assert.Equal("", test);
+        }
+
+        [Fact]
+        public void GetDirectoryOrRootName_FileExists()
+        {
+            string exePath = AppDomain.CurrentDomain.BaseDirectory;
+            Stream fileStream = File.Create(TEST_FILE_NAME);
```

Comment:
If you're going to be creating files or directories in a test, please use `TempFileBase` or `TempFile` and `TempDirectory`. For example

``` c#
               using (var testDirectory = new TempDirectory(GetTestFilePath()))
               using (var file = new TempFile(Path.Combine(testDirectory.Path, "file")))
...
```

Certainly "c:" won't work on Unix...